foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/dao/Dao.java

```diff
@@ -0,0 +1,43 @@
+package com.boozallen.aissemble.configuration.dao;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Configuration::Store
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.configuration.exception.PropertyDaoException;
+
+import java.util.Set;
+
+public interface Dao<T> {
+
+    boolean checkEmpty();
+
+    /**
+     * Read T from Vault with given group name and property name
+     * @param groupName group name
+     * @param propertyName property name
+     * @return T
+     */
+    T read(String groupName, String propertyName);
```

Comment:
Q: I'd originally thought this was an interface for _any_ DAO we may need for the config store but this seems specific to properties.  If it's only an interface for properties, why the type parameter? Do we anticipate Property sub-types?

Reply:
It should be generic. yeah.. I think so. From the design we do have another LimitationMetaDataDao, which has the groupName and propertyName as well. ref: https://confluence.boozallencsn.com/display/AIOPS/Universal+Configurations?preview=/443073804/458738303/class-diagram.drawio.png